components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
 
 			switch(i) {
 				case 'name':
-					values.personaName = settings[i];
+					out[8] = {
+						"name": formd[8].name,
```

Comment:
What if field order ever changes?

Reply:
Longer code is better than broken code.

Reply:
I can add that it doesn't use the default values on the page, but creates them itself and copies over the correct values.